docs/docker/DHCP.md

```diff
@@ -0,0 +1,50 @@
+---
+title: Docker DHCP and Network Modes
+description: Setting up DHCP for Docker Pi-hole
+last_updated: Sat Feb 09 00:00:00 2019 UTC
+---
+
+# Docker DHCP and Network Modes
+
+Docker runs in a separate network by default called a docker bridge network, which makes DHCP want to serve addresses to that network and not your LAN network where you probably want it. This document details why Docker Pi-hole DHCP is different from normal Pi-hole and how to fix the problem.
+
+### Technical details
+
+Docker's bridge network mode is default and recommended as a more secure setting for containers because docker is all about isolation, they isolate processes by default and the bridge network isolates the networking by default too. You gain access to the isolated container's service ports by using port forwards in your container's runtime config; for example `-p 67:67` is DHCP. However DHCP protocol operates through a network 'broadcast' which cannot span multiple networks (docker's bridge, and your LAN network). In order to get DHCP on to your network there are a few approaches:
+
+## Working network modes
+
+Here are details on setting up DHCP for Docker Pi-hole for various network modes available in docker.
+
+### Docker Pi-hole with host networking mode
+
+Possibly the simplest way to get DHCP working with Docker Pi-hole is to use [host networking](https://docs.docker.com/network/host/) which makes the container be on your LAN Network like a regular Raspberry Pi-hole would be, allowing it to broadcast DHCP. It will have the same IP as your Docker host server in this mode so you may still have to deal with port conflicts.
+
+- Inside your docker-compose.yml remove all ports and replace them with: `network_mode: host`
+- `docker run --net=host` if you don't use docker-compose
+
+**Advantages**: Simple, easy, and fast setup
```

Comment:
Can you put these right below each header that they reference? It may be confusing if they are right above a header that they do not reference, and moving it would make skimming the article easier.